File: twitter_data_tool/filter_celebrities.py
#!/usr/bin/env python

from sys import argv

c = set()
with open("celebrities_id.txt") as f:
    for line in f:
        id = eval(line)
        c.add(id)

with open(argv[1]) as f:
    with open(argv[2]) as fout:
        for line in argv[1]:
            x, y = map(lambda x: eval(x), line.split('\t'))
            if x in c:
                fout.write(x)
    

File: twitter_data_tool/filter_celebrities.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <cstdlib>
#include <unordered_set>
using namespace std;

int main(int argc, char *argv[]) {
    
    if (argc < 3) {
        cout << "usage: " << argv[0] << " <infile> <outfile>" << endl;
        return 0;
    }

    const char *infile = argv[1];
    const char *outfile = argv[2];

    unordered_set<int> c(8000);
    ifstream cfin("celebrities_id.txt");
    string line;
    while (getline(cfin, line), !line.empty()) {
        int id = stoi(line);
        c.insert(id);
    }
    cfin.close();

    cout << "dict read" << endl;

    ifstream fin(infile);
    ofstream fout(outfile);
    
    while (getline(fin, line), !line.empty()) {
        int x = stoi(line);
        if (c.find(x) != c.end()) {
            fout << line << endl;
        }
    }

    fout.close();
    fin.close();

    return 0;
}
